driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -285,6 +285,13 @@ private PoolingOptions options() {
 
         SettableFuture<Connection> future = SettableFuture.create();
         pendingBorrows.add(future);
+
+        // If we raced with shutdown, make sure the future will be completed. This has no effect if it was properly
+        // handled in closeAsync.
+        if (phase.get() == Phase.CLOSING) {
```

Comment:
Nice, for some reason I thought handling this would be much more complex, but checking the state of the pool after adding the future makes perfect sense.